### DISCUSSION TITLE
etcdctl: change the default `write-out`

### DISCUSSION THREAD

**Discussion by till** [etcd-io/etcd#14686] [Ideas]
Hi —

have to admit I am not using `etcdctl` on a daily basis (and I had not planned to). Just using it by means of k8s.

Would it be acceptable to either change the default `write-out` to something that people can read (probably a BC break)? Or would the project accept a PR to add an env variable or a dotfile (`~/.etcdctl`) to be able to set the `write-out` for the user? `simple` as a default just seems way too simple for people. And the other requires lots of typing when things get heated.

`write-out` for me is just the tip of the 🧊 iceberg. Hence, maybe a config file for defaults would be nice?

Cheers!
Till

**Comment by serathius** [comment]
Non breaking alternative would be to add support for `ETCDCTL_WRITEOUT` environment that you could set in your `.bashrc`.

**Reply by till** [reply]
@serathius are you saying there is an `ETCDCTL_WRITEOUT` or are you suggesting what I suggested?

**Reply by serathius** [reply]
There is no `ETCDCTL_WRITEOUT` now, but contributions are welcomed.

**Reply by till** [reply]
@serathius I had a quick look and I can probably make it work with Cobra, but I guess one last question, I saw that there should be other environment variables already I wanted to piggy back off of that:

```
❯ fgrep -r 'ETCDCTL_CERT' *
etcdctl/READMEv2.md:+ env variable: ETCDCTL_CERT_FILE
etcdctl/README.md:ETCDCTL_CERT=/tmp/cert.pem
```

Is this hidden in another lib used by this project? Or should I ignore this and just add `ETCDCTL_WRITEOUT` here?

**Comment by ahrtr** [comment]
etcdctl already supports `ETCDCTL_WRITE_OUT`. 
FYI. https://github.com/etcd-io/etcd/pull/14699#issuecomment-1307127781

**Reply by till** [reply]
Le sigh. How do I find what other variables are supported? I tried searching, but found nothing. When I look at the code, is it save to assume that everything prefixed with `ECTDCTL` and some how matches an option name can be set?

**Reply by ahrtr** [reply]
Usually `etcdctl` already supports env variables something like "ETCDCTL_" + ${FLAG_NAME}. For example, `etcdctl` supports flag "write-out", then it supports "ETCDCTL_WRITE_OUT" automatically.  Please read [etcdctl/ctlv3/command/global.go#L103](https://github.com/etcd-io/etcd/blob/3e903d0b12e399519a4013c52d4635ec8bdd6863/etcdctl/ctlv3/command/global.go#L103)

**Reply by till** [reply]
@ahrtr Yeah, I read the code. And I found spf13/cobra#793

I think I never used cobra and this was one of the reasons. Kong and urfave/cli have that already. 

Would you accept a PR to the docs highlighting that? Maybe this page: https://etcd.io/docs/v3.5/dev-guide/interacting_v3/
